### PR TITLE
Expose variables

### PR DESCRIPTION
**Pull request by iandday**
Set all customization variables in .env file via environmental variables

#84

### CODE REVIEW

**Comment by InputObject2** [comment]
Instead of doing variables individually, why not 

`RUN env > /var/www/html/.env` ?

**Comment by iandday** [comment]
You could do that, but I think that would require rebuilding the container. Using env variables will be more streamlined once you get the image building and pushing to GitHub’s container registry working. That way there won’t be a need to clone the repo or build the image. Keeping everything in the docker-compose file makes life easier for people who keep their compose files in version control, everything needed is in one file.

**Comment by InputObject2** [comment]
I should have double checked what I wrote before commenting, my bad.

You're right, I meant in the docker-entrypoint.sh script for use at runtime, no real point putting those in the dockerfile at build time.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -29,35 +29,35 @@ create_environment_file() {
     # App settings
```
Comment (InputObject2) [inline docker-entrypoint.sh:29]: May I suggest to use a herefile to make it more easily copy-pastable?

```bash
cat <<EOF >> /var/www/html/.env
# App settings
APP_NAME="HortusFox"
APP_DEBUG=$APP_DEBUG
APP_BASEDIR="$APP_BASE_DIR"
APP_LANG="$APP_LANG"
APP_WORKSPACE="$APP_WORKSPACE"
APP_ENABLESCROLLER="$APP_ENABLE_SCROLLER"
APP_OVERLAYALPHA="$APP_OVERLAY_ALPHA"
APP_ENABLECHAT="$APP_ENABLE_CHAT"
APP_ONLINEMINUTELIMIT="$APP_ONLINE_MINUTE_LIMIT"
APP_SHOWCHATONLINEUSERS="$APP_SHOW_CHAT_ONLINE_USERS"
APP_SHOWCHATTYPINGINDICATOR="$APP_SHOW_CHAT_TYPING_INDICATOR"
APP_OVERDUETASK_HOURS=$APP_OVERDUE_TASK_HOURS
APP_CRONPW="$APP_CRON_PW"
APP_CRONJOB_MAILLIMIT=$APP_CRON_MAIL_LIMIT
APP_GITHUB_URL="https://github.com/danielbrendel/hortusfox-web"
APP_SERVICE_URL="https://www.hortusfox.com"
APP_ENABLEHISTORY=$APP_ENABLE_HISTORY
APP_HISTORY_NAME="$APP_HISTORY_NAME"

# Session
SESSION_ENABLE=true
SESSION_DURATION=31536000
SESSION_NAME=null

# Photo resize factors
PHOTO_RESIZE_FACTOR_DEFAULT=$PHOTO_RESIZE_FACTOR_DEFAULT
PHOTO_RESIZE_FACTOR_1=$PHOTO_RESIZE_FACTOR_1
PHOTO_RESIZE_FACTOR_2=$PHOTO_RESIZE_FACTOR_2
PHOTO_RESIZE_FACTOR_3=$PHOTO_RESIZE_FACTOR_3
PHOTO_RESIZE_FACTOR_4=$PHOTO_RESIZE_FACTOR_4
PHOTO_RESIZE_FACTOR_5=$PHOTO_RESIZE_FACTOR_5

# Database settings
DB_ENABLE=true
SMTP_PORT=$SMTP_PORT
SMTP_USERNAME="$SMTP_USERNAME"
SMTP_PASSWORD="$SMTP_PASSWORD"
SMTP_ENCRYPTION="$SMTP_ENCRYPTION"

# Logging
LOG_ENABLE=$LOG_ENABLE
EOF

```

Comment (danielbrendel) [reply docker-entrypoint.sh:29]: Looks more clean and better maintainable.

Comment (iandday) [reply docker-entrypoint.sh:31]: I agree, I updated the entrypoint script and took a stab at updated the variable documentation in readme.md

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -142,27 +145,15 @@ set_apache_server_name
...
-if [ $(mysql -u "$DB_USERNAME" -p"$DB_PASSWORD" -h "$DB_HOST" -D "$DB_DATABASE" -N -s -e "SELECT COUNT(*) FROM information_schema.TABLES WHERE TABLE_SCHEMA = '$DB_DATABASE' AND TABLE_NAME = 'plants';") -eq 1 ]
-then
-    echo "Running unapplied database migrations..."
-    php asatru migrate:list
-else
-    echo "Running full database migrations..."
-    php asatru migrate:fresh
-fi
+echo "Running database migrations..."
+php asatru migrate:fresh
```
Comment (aeber) [inline docker-entrypoint.sh:149]: Hi! I'm not sure if it's a good idea to run `migrate:fresh` on every start of the container as this seems to drop and recreate all tables? Doesn't this lead to dataloss?
See `migrate_fresh()` [here](https://github.com/danielbrendel/dnyAsatruPHP-Framework/blob/410b2fe4710bce9c4e5f662e640f12b864e5cbfb/src/database.php#L1205), `dropAll()` [here](https://github.com/danielbrendel/dnyAsatruPHP-Framework/blob/410b2fe4710bce9c4e5f662e640f12b864e5cbfb/src/database.php#L1139) and the `down()` functions of the migrations in `/app/migrations`.

Comment (iandday) [reply docker-entrypoint.sh]: I agree! Ironically enough i was the one to propose the change initially.  I've reverted the deletion.

**Pull request merged**